File: Dagger2TestAndroid/app/src/main/java/com/example/raghu/dagger2testandroid/presenter/MainPresenterContract.java
package com.example.raghu.dagger2testandroid.presenter;

import android.support.annotation.Nullable;

/**
 * Created by raghu on 29/7/17.
 */

public interface MainPresenterContract {


    interface View {

        void showData(String message);




    }

    interface Presenter  {

        void doSomething();
    }
}

File: SpecBeeMapAssignment/app/src/main/java/com/example/raghu/specbeemapassignment/presenter/MainPresenterContract.kt
package com.example.raghu.dagger2testandroid.presenter

import com.example.raghu.specbeemapassignment.models.Example

/**
 * Created by raghu on 29/7/17.
 */

interface MainPresenterContract {


    interface View {

        fun showData(example: Example)
        fun showDataWeather(example: com.example.raghu.specbeemapassignment.models.weather.Example)


    }

    interface Presenter {

        fun doSomething(query:String,key:String)
        fun getWeatherData(lat:Double,lng:Double,APPID:String)

        fun unSubscribe()
    }
}

